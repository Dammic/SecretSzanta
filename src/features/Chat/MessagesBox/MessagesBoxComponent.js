'use strict'
import React from 'react'

const MessagesBoxComponent = ({
    messages = []
}) => {
    const buildMessage = (message) => {
        const {time, author, content, messageIndex} = message
        return (
            <div key={messageIndex} className="message">
                [{time}] {author === 'SERVER' ? '' : author}: {content}
            </div>
        )
    }

    return (
        <div className="messages-box">
            {messages.map((message) => buildMessage(message))}
        </div>
    )
}

export default MessagesBoxComponent


