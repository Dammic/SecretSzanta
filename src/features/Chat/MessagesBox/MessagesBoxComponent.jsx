import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import classNames from 'classnames/bind'

import styles from './MessagesBox.css'

const buildMessage = (message, index) => {
    const { time, author, content } = message
    const displayedMessage = `[${time}]${author ? ` ${author}:` : ':'} ${content}`

    return (
        <div key={index} className={classNames(styles.message, { [styles.serverMessage]: !author })}>
            {displayedMessage}
        </div>
    )
}

const MessagesBoxComponent = React.forwardRef(({
    messages = [],
}, messagesBoxRef) => (
    <div className={styles.messagesBox} ref={messagesBoxRef}>
        {map(messages, (message, index) => buildMessage(message, index))}
    </div>
))

MessagesBoxComponent.displayName = 'MessagesBoxComponent'
MessagesBoxComponent.propTypes = {
    messages: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
}
export default MessagesBoxComponent

