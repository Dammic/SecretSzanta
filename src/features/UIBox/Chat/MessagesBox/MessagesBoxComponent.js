'use strict'
import React from 'react'
import classNames from 'classnames/bind'

const MessagesBoxComponent = ({
    messages = [],
    setMessagesEndRef
}) => {
    const buildMessage = (message, index) => {
        const {time, author, content} = message
        return (
            <div key={index} className={classNames("message", {'server-message': !author})}>
                [{time}]{author ? ` ${author}: ` : ':'} {content}
            </div>
        )
    }

    return (
        <div className="messages-box">
            {messages.map((message, index) => buildMessage(message, index))}
            <div id="EndOfMessages" ref={setMessagesEndRef} />
        </div>
    )
}

export default MessagesBoxComponent

