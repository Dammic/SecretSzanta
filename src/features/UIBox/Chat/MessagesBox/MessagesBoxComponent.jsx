import React from 'react'
import PropTypes from 'prop-types'
import { map } from 'lodash'
import classNames from 'classnames/bind'

import styles from './MessagesBox.css'

const buildMessage = (message, index) => {
    const { time, author, content } = message
    const displayedMessage = `[${time}]${author ? ` ${author}: ` : ':'} ${content}`

    return (
        <div key={index} className={classNames(styles.message, { [styles.serverMessage]: !author })}>
            {displayedMessage}
        </div>
    )
}

const MessagesBoxComponent = ({
    messages = [],
    setMessagesBoxRef,
}) => (
    <div className={styles.messagesBox} ref={setMessagesBoxRef}>
        {map(messages, (message, index) => buildMessage(message, index))}
    </div>
)

MessagesBoxComponent.propTypes = {
    messages: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
    setMessagesBoxRef: PropTypes.func,
}
export default MessagesBoxComponent

