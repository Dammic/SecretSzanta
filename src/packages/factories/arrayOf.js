import faker from 'faker'
import { times } from 'lodash'

export const arrayOf = (callback, amount, options = { seed: null }) => {
    if (options.seed) {
        faker.seed(options.seed)
    }

    const result = []
    times(amount, () => result.push(callback()))
    return result
}
