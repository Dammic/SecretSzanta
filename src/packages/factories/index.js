export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
