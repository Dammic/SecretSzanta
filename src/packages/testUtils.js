import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
