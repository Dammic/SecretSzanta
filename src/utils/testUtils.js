import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
