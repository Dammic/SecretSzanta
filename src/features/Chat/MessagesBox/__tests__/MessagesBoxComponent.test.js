import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import { aMessage, arrayOf } from 'packages/factories'
import MessagesBoxComponent from '../MessagesBoxComponent'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        setMessagesBoxRef: jest.fn(),
        messages: arrayOf(aMessage, 3, { seed: 1 }),
        ...propsOverrides,
    }
    const component = renderMethod(<MessagesBoxComponent { ...props } />)
    return { props, component }
}

describe('<MessagesBoxComponent />', () => {
    it('renders 3 messages if there are any passed', () => {
        const { props } = setupProps()
        expectMatchingSnapshot(<MessagesBoxComponent {...props} />)
    })

    it('renders server messages', () => {
        const { props } = setupProps({
            messages: [...arrayOf(aMessage, 2, { seed: 1 }), aMessage({ author: null }, { seed: 1 })],
        })
        expectMatchingSnapshot(<MessagesBoxComponent {...props} />)
    })

    it('renders no messages if no messages have been passed', () => {
        const { props } = setupProps({ messages: null })
        expectMatchingSnapshot(<MessagesBoxComponent {...props} />)
    })
})
