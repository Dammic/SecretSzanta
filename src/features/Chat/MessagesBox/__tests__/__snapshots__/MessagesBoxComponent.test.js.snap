// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesBoxComponent /> renders 3 messages if there are any passed 1`] = `
<div
  className="messagesBox"
>
  <div
    className="message"
  >
    [10:59] Mina12: Liaison
  </div>
  <div
    className="message"
  >
    [5:05] Grace.Dibbert: matrix Suriname e-tailers
  </div>
  <div
    className="message"
  >
    [16:31] Connor.Kihn2: European Unit of Account 17(E.U.A.-17) vortals
  </div>
  <div
    className="message serverMessage"
  >
    [10:59]: Liaison
  </div>
</div>
`;

exports[`<MessagesBoxComponent /> renders no messages if no messages have been passed 1`] = `
<div
  className="messagesBox"
/>
`;
